app/code/Magento/Theme/Model/Menu/GetPredefinedMenu.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Theme\Model\Menu;
+
+use Magento\Theme\Api\GetMenuInterface;
+use Magento\Theme\Api\Data\MenuNodeInterface;
+
+/**
+ * GetMenuInterface implementation which returns menu items passed during instantiation.
+ *
+ * @api
+ */
+class GetPredefinedMenu implements GetMenuInterface
+{
+    /**
+     * @var MenuNodeInterface[]
+     */
+    private $menuNodes;
+
+    /**
+     * GetPredefinedMenu constructor.
+     *
+     * @param array $menuNodes
+     */
+    public function __construct(array $menuNodes = [])
+    {
+        $this->menuNodes = array_map(
+            function (MenuNodeInterface $menuNode) {
+                return $menuNode;
+            },
+            $menuNodes
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(): array
```

Comment:
Command naming approach is a bit inconsistent with other Magento 2 classes, so I would suggest renaming the method to "getMenuNodes" (as it actually is a getter and does not execute anything), also I'd avoid action class naming i.e. use naming like MenuNodesProvider, MenuProvidersPool instead of Get...